core/client/fs/src/main/java/alluxio/client/block/stream/BlockInStream.java

```diff
@@ -337,8 +341,42 @@ public void seek(long pos) throws IOException {
     }
     if (pos < mPos) {
       mEOF = false;
+      if (mDataReader instanceof NaiveSharedGrpcDataReader) {
+        NaiveSharedGrpcDataReader reader = (NaiveSharedGrpcDataReader)mDataReader;
+        reader.seek(pos);
+        if (mCurrentChunk != null) {
+          mCurrentChunk.release();
+          mCurrentChunk = null;
+        }
+      } else {
+        closeDataReader();
+      }
+    } else if (pos < mLength) {
+      // Try to read data already in queue
+      long curPos = mPos;
+      while (mCurrentChunk != null && curPos < pos) {
+        long nextPos = curPos + mCurrentChunk.readableBytes();
+        if (nextPos <= pos) {
+          curPos = nextPos; 
+          mCurrentChunk.release();
+          mCurrentChunk = mDataReader.readChunkNoWait();
+        } else {
+          // TODO introduce seek in DataBuffer
+          int toRead = (int)(pos - curPos);
+          final byte[] b = new byte[toRead];
+          mCurrentChunk.readBytes(b, 0, toRead);
+          curPos = pos;
+        }
+      }
+
+      if (curPos < pos) {
+        // Not enough data in queue, close the data reader
+        closeDataReader();
+      }
+    } else {
+      closeDataReader();
```

Comment:
I think this is a redundant case. please check

Reply:
Removed this block, think it's abundant as well